src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -90,12 +95,47 @@ public static ExtensionSettings getExtensionSettings() throws IOException {
     PluginResponse handlePluginsRequest(PluginRequest pluginRequest) {
         logger.info("Handling Plugins Request");
         PluginResponse pluginResponse = new PluginResponse("RealExtension");
+        opensearchNode = pluginRequest.getSourceNode();
         return pluginResponse;
     }
 
-    IndicesModuleResponse handleIndicesModuleRequest(IndicesModuleRequest indicesModuleRequest) {
+    IndicesModuleResponse handleIndicesModuleRequest(IndicesModuleRequest indicesModuleRequest, TransportService transportService) {
         logger.info("Indices Module Request");
         IndicesModuleResponse indicesModuleResponse = new IndicesModuleResponse(true, true, true);
+
+        // CreateComponent
+        transportService.connectToNode(opensearchNode);
+        final CountDownLatch inProgressLatch = new CountDownLatch(1);
+        try {
+            logger.info("Sending request to opensearch");
+            ClusterStateResponseHandler clusterStateResponseHandler = new ClusterStateResponseHandler();
+            transportService.sendRequest(
+                opensearchNode,
+                ExtensionsOrchestrator.REQUEST_EXTENSION_CLUSTER_STATE,
+                new ExtensionRequest(ExtensionsOrchestrator.RequestType.REQUEST_EXTENSION_CLUSTER_STATE),
+                clusterStateResponseHandler
+            );
+            ClusterSettingResponseHandler clusterSettingResponseHandler = new ClusterSettingResponseHandler();
+            transportService.sendRequest(
+                opensearchNode,
+                ExtensionsOrchestrator.REQUEST_EXTENSION_CLUSTER_SETTINGS,
+                new ExtensionRequest(ExtensionsOrchestrator.RequestType.REQUEST_EXTENSION_CLUSTER_SETTINGS),
+                clusterSettingResponseHandler
+            );
+            LocalNodeResponseHandler localNodeResponseHandler = new LocalNodeResponseHandler();
+            transportService.sendRequest(
+                opensearchNode,
+                ExtensionsOrchestrator.REQUEST_EXTENSION_LOCAL_NODE,
+                new ExtensionRequest(ExtensionsOrchestrator.RequestType.REQUEST_EXTENSION_LOCAL_NODE),
+                localNodeResponseHandler
+            );
+            inProgressLatch.await(1, TimeUnit.SECONDS);
```

Comment:
Looks like this CountDownLatch is only used to wait 1 second, and it isn't counted down.  The latch should be updated when the responses come back, I would suggest the count is 3 and its expected each of the `sendRequest` methods are responsible for lowering the count by one.

I would recommend reauthoring this code to leverage the collections objects from future, `allOf(....)`.  If this doesn't work it might be worthwhile to invest in writing helpers to make declaring code following this pattern smaller and more error resistance since I think this example will be replicated in many places.
```java
CompletableFuture<?> clusterStateRequest =  transportServiceProvider.startRequest(....);
CompletableFuture<?> clusterSettingsRequest =  transportServiceProvider.startRequest(....);
CompletableFuture<?> localNodeRequest =  transportServiceProvider.startRequest(....);
try {
   // Result isn't used, timeout or inner exceptions are handled in catch block
   CompletableFuture.allOf(clusterStateRequest, clusterSettingsRequest, localNodeRequest).get(1, TimeUnit.SECONDS).
} catch (Exception e) {
   logger.error(e);
}
```

Reply:
Removed CountDown Latch as it was not needed here.

Reply:
I'm confused, don't this method need to wait until the transport requests have finished?  If we remove a mechanism to wait we aren't able to handle failures where the extension wasn't able to honor one of the requests and send back error information.

If these requests are executed synchronously, I suppose the current code make sense, but it slows down the response considerable if any of these actions can be done in parallel.  Since none of the responses are used as the input to later requests this seems to be the case.

Reply:
The latch was added to make the calls asyn before but since now each request has their own handler they have their own latch and can be handled separately. @saratvemulapalli correct me I'm wrong here

Reply:
Removed the latch here as it was a hacky way of making Transport call and from the handler as well.